src/Analyser/MutatingScope.php

```diff
@@ -2347,7 +2347,11 @@ static function (Node $node, Scope $scope) use ($arrowScope, &$arrowFunctionImpu
 					/** @var non-empty-string $name */
 					$name = $node->name->value;
 					$functionName = new Name($name);
-				} elseif ($node->name instanceof FuncCall && $node->name->name instanceof Name) {
+				} elseif (
+					$node->name instanceof FuncCall
+					&& $node->name->name instanceof Name
+					&& $node->name->isFirstClassCallable()
```

Comment:
Seems like the purpose of this if was to resolve node like
`strtoupper(...)($foo)` 
which is resolved to `strtoupper($foo)`
but it messed up with `foo($bar)($baz)`
which was resolved to `foo($baz)`.